python/cudf_polars/cudf_polars/experimental/shuffle.py

```diff
@@ -192,3 +192,34 @@ def _(
         partition_info[ir.children[0]].count,
         partition_info[ir].count,
     )
+
+
+def _maybe_shuffle_frame(
+    frame: IR,
+    on: tuple[NamedExpr, ...],
+    partition_info: MutableMapping[IR, PartitionInfo],
+    config_options: ConfigOptions,
+    output_count: int,
+) -> IR:
+    from cudf_polars.experimental.parallel import PartitionInfo
```

Comment:
This method is moved from `_join.py` to here. It'd be nice to move this import out of here (currently getting an import error from a cycle), maybe by defining `PartitionInfo.base`

Reply:
Shouldn't this be:

```python
from cudf_polars.experimental.base import PartitionInfo
```